Assets/Scripts/PlayerMovement.cs

```diff
@@ -52,11 +56,46 @@ private void UpdateCameraMovement()
         Camera.transform.localEulerAngles = new Vector3(-_pitch, 0, 0);
     }
 
-    private void UpdateAttack()
+    private void UpdateAttack() // update attack function, which checks for attacks and what direction
     {
-        if (Input.GetKeyDown(KeyCode.Mouse0))
+        
+        if (Input.GetKeyDown(KeyCode.Mouse0)) // if mouse button is pressed
+        {
+            _holdMouseDown = true; // then mouse is being hold
+            Cursor.lockState = CursorLockMode.None; // mouse should be moved freely
+            Cursor.visible = false; // but not visible to the player
+        }
+            
+        if (Input.GetKeyUp(KeyCode.Mouse0) && _holdMouseDown == true && _entity.LoseStamina(10)) // if mouse button let go, normal attack cost 10 stamina
         {
-            _entity.Attack();
+            _holdMouseDown = false; // then mouse is not holding down anymore
+            Vector2 MousePos = Input.mousePosition; // check the new pos for direction
+            Cursor.lockState = CursorLockMode.Locked; // set mouse locked agin
+            Vector2 MousePosCenter = new Vector2(Screen.width / 2 , (Screen.height / 2) + 20);
+            Debug.Log(Direction(MousePos, MousePosCenter)); // put in console direction
+            _entity.Attack(); // attack
+            //We could change the attack functions to set the number of stamina in there of how much stamina it costs
         }
     }
+
+    private string Direction(Vector2 New_Pos, Vector2 MousePosCenter) // Functio  to check direction
+    {
+        float diffx = New_Pos.x - MousePosCenter.x; // check distance x
+        float diffy = New_Pos.y - MousePosCenter.y; // check distance y
+
+        if (diffx > 50 && diffy < 50 && diffy > -50) // right
+            return "Right";                   //                                                 ----------------
```

Comment:
Ik denk dat we van de returnwaardes uiteindelijk beter een enum kunnen maken. Dat verkleint de kans dat er iets misgaat tijdens het vergelijken bij andere functies

Reply:
Naar enums veranderd, beter?